src/theory/strings/theory_strings_rewriter.cpp

```diff
@@ -1789,6 +1954,120 @@ Node TheoryStringsRewriter::collectConstantStringAt( std::vector< Node >& vec, u
   }
 }
 
+bool TheoryStringsRewriter::stripSymbolicLength(std::vector<Node>& n1,
+                                                std::vector<Node>& nr,
+                                                int dir,
+                                                Node& curr)
+{
+  Assert(dir == 1 || dir == -1);
+  Assert(nr.empty());
+  Node zero = NodeManager::currentNM()->mkConst(CVC4::Rational(0));
+  bool ret = false;
+  bool success;
+  unsigned sindex = 0;
+  do
+  {
+    Assert(!curr.isNull());
+    success = false;
+    if (curr != zero && sindex < n1.size())
+    {
+      unsigned sindex_use = dir == 1 ? sindex : ((n1.size() - 1) - sindex);
+      if (n1[sindex_use].isConst())
+      {
+        // could strip part of a constant
+        Node lowerBound = getConstantArithBound(curr);
+        if (!lowerBound.isNull())
+        {
+          Assert(lowerBound.isConst());
+          Rational lbr = lowerBound.getConst<Rational>();
+          if (lbr.sgn() > 0)
+          {
+            Assert(checkEntailArith(curr));
```

Comment:
1. Shouldn't this use the strict version?
2. Since we don't currently have any guaranteed relationship between `checkEntailArith` and `getConstantArithBound > 0` this could fail unexpectedly. Should we maybe have this kind of assert in `getConstantArithBound` itself (or even call `checkEntailArith` as discussed offline)? Then we would at least check this assertion more frequently and document the relationship.

Reply:
Yes, this is a good idea, I'll add an assertion.

Reply:
What about using the strict version of `checkEntailArith` here?